recipes/google-cloud-cpp/2.x/conanfile.py

```diff
@@ -0,0 +1,333 @@
+        #   https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B
+        check_min_vs(self, "192")
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} shared not supported by Visual Studio")
+
+        if hasattr(self, "settings_build") and cross_building(self):
+            raise ConanInvalidConfiguration(
+                "Recipe not prepared for cross-building (yet)"
+            )
+
+        if (
+            self.settings.compiler == "clang"
+            and Version(self.settings.compiler.version) < "6.0"
+        ):
+            raise ConanInvalidConfiguration("Clang version must be at least 6.0.")
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 14)
+
+        if (
+            self.settings.compiler == "gcc"
+            and Version(self.settings.compiler.version) < "5.4"
+        ):
+            raise ConanInvalidConfiguration("Building requires GCC >= 5.4")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def requirements(self):
+        self.requires("protobuf/3.21.9")
+        self.requires("grpc/1.50.1")
+        self.requires("nlohmann_json/3.10.0")
+        self.requires("crc32c/1.1.1")
+        self.requires("abseil/20220623.0")
+        self.requires("libcurl/7.88.1")
+        self.requires("openssl/1.1.1t")
+        self.requires("zlib/1.2.13")
+        # `google-cloud-cpp` contains code generated from the proto files.
+        # Working with older versions of these protos almost always will fail, as
+        # at least some of the RPCs included in the GRPC-generator stubs will be
+        # missing.
+        # Working with newer versions of these protos almost always will work. There
+        # are very few breaking changes on the proto files.
+        self.requires(f"googleapis/{self._GOOGLEAPIS_VERSIONS[self.version]}")
+
+    def build_requirements(self):
+        # For the grpc-cpp-plugin executable
+        self.tool_requires("grpc/1.50.1")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["CMAKE_PROJECT_google-cloud-cpp_INCLUDE"] = os.path.join(self.source_folder, "conan_cmake_project_include.cmake")
+        tc.variables["BUILD_TESTING"] = False
+        tc.variables["GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK"] = False
+        tc.variables["GOOGLE_CLOUD_CPP_ENABLE"] = ",".join(self._components())
+        tc.generate()
+        # We may need a build environment to run the gRPC plugin. I (@coryan)
+        # took this code for the gRPC test packages. They may not be the "Right Thing"[tm]
+        # in this context.
+        if hasattr(self, "settings_build"):
+            VirtualBuildEnv(self).generate()
+        else:
+            # Cover case in Conan 1.x when only one profile is provided
```

Comment:
This is deprecated and does not need to be supported, we can remove it if you dont need it :)

Reply:
Candidly I am not sure if I need it.  Local testing with v2 seems to require a `VirtualRunEnv()` call, but I may be doing something else wrong.  I would appreciate any guidance here.